### ISSUE TITLE
Swift Type check and functions build times not generated

### ISSUE THREAD

**Issue by stherold** [MobileNativeFoundation/XCLogParser#190]
Hey there 👋 

I don't get any compilation time statistics in teh HTML file. According to [this post](https://stackoverflow.com/a/69393293/971329) the Swift compiler flags the generated website suggests don't exist anymore:

![Screenshot 2023-10-24 at 18 05 39](https://github.com/MobileNativeFoundation/XCLogParser/assets/64915710/15ce7236-db37-4fd3-bc30-76b8a9ac39a3)

**Comment by david-bandlab** [comment]
Same here 🙋‍♂️

**Comment by Bobbyphtr** [comment]
Same here

**Comment by BalestraPatrick** [comment]
Have you tried the new flags suggested at the link to see if they work? If they do, then we welcome a contribution to change the wording and references to those now removed flags.

**Comment by hoangatuan** [comment]
@stherold What is the xcodebuild command you was using?

**Comment by hoangatuan** [comment]
@stherold 
Make sure you add these flags to "Other Swift Flags" in your project build setting
 "-Xfrontend", 
 "-warn-long-function-bodies=200",
 "-Xfrontend",
 "-warn-long-expression-type-checking=200"

If your workspace uses swift package manager, can insert this code to your Package.swift file

```swift
for target in package.targets {
   target.swiftSettings = target.swiftSettings ?? []
   target.swiftSettings?.append(
       .unsafeFlags([
           "-Xfrontend", "-warn-long-function-bodies=\(config.warnLongFunctionBodies)", "-Xfrontend", "-warn-long-expression-type-checking=\(config.warnLongExpressionTypeChecking)",
        ])
    )
}
```

**Comment by hoangatuan** [comment]
@stherold 
For Xcode 15.3, seems like XCLogParse is breaking.
For now you can use this script to get warnings

```bash
xcodebuild -workspace <ws-name>.xcworkspace -scheme <scheme> -destination "platform=iOS Simulator,name=iPhone 15 Pro,OS=latest" OTHER_SWIFT_FLAGS="-Xfrontend -warn-long-function-bodies=200 -Xfrontend -warn-long-expression-type-checking=200" clean build | grep .[0-9]ms | grep -v ^0.[0-9]ms | grep "^$ROOT_DIRECTORY" | sort -nr > \(warningsFileName)
```

**Comment by stherold** [comment]
Yes that generates some output 👍

**Comment by Blackjacx** [comment]
Yes that generates some output 👍 Does that mean that it would be part of the final report?

**Comment by hoangatuan** [comment]
@Blackjacx 
Sorry I dont understand your question "Does that mean that it would be part of the final report?".

This is just a simple script that extract Xcode build logs directly by using regex. It doesn't use XCLogParser, so it can't generate the html report

**Comment by hbanzon** [comment]
Getting this issue as well when updating to Xcode 15.3.  Any ETA on the fix?

**Comment by ignindra** [comment]
Does anyone manage to make it work with any version of Xcode newer than 15.3? I'm on 15.4 and I'm in the same situation as stherold's screenshot in the first post.

Was wondering whether I should downgrade below 15.3 or upgrade my Xcode.

**Comment by Iron-Ham** [comment]
I decided to look into why this is. Will update this comment as I continue investigating.

I started by generating a valid `xcactivitylog` output. I made that `xcactivitylog` human readable by executing:

```
gzcat /path/to/file.xcactivitylog | tr '\r' '\n' > human_readable.log
```

I ensured that the timing data is present and accounted for, and I even placed it directly into the existing unit tests to make sure that we're getting the outputs there as expected. All is working fine.

Tracing it back up, I found the first point of breakage:

https://github.com/MobileNativeFoundation/XCLogParser/blob/7cf8c50e1b84effcda573bbf8b26bdd0a6b4614f/Sources/XCLogParser/parser/ParserBuildSteps.swift#L197

Here, we check to see if we have a swift compilation step. If we do, we will add the log section and continue on. Note that the compile parser owns the function timing and type checking parsers. Herein lies the issue: in modern Xcode versions, we will **never** recognize a swift compilation step. 

We identify a swift compilation step here:

https://github.com/MobileNativeFoundation/XCLogParser/blob/7cf8c50e1b84effcda573bbf8b26bdd0a6b4614f/Sources/XCLogParser/parser/BuildStep.swift#L99-L100

However, that's not how swift compilation steps are defined currently. They are not `CompileSwift` – but rather, `SwiftCompile`:

```
"SwiftCompile normal arm64 Compiling\ Builder.swift,\ Functions.swift,\ UIActivityIndicatorViewStyles.swift,\ UIButtonStyles.swift,\ UIFont+MutableTraits.swift,\ UIImageViewStyles.swift,\ UILabelStyles.swift,\ UIScrollViewStyles.swift,\ UIStackViewStyles.swift,\ UITextFieldStyles.swift,\ UITextViewStyles.swift,\ UIViewStyles.swift /<REDACTED> (in target 'Palette' from project '<REDACTED>')
```

I'm going to rebuild with this change and update.
